desc/geometry/curve.py

```diff
@@ -817,8 +820,21 @@ def __init__(
             errorif(knots[-1] > 2 * np.pi, ValueError, "knots must lie in [0, 2pi]")
             knots = knots[:-1] if closed_flag else knots
 
+        if discontinuous_indices is None:
+            interval_indices = [[]]
+        else:
+            # check that input is monotonic
+            assert np.all(
+                discontinuous_indices == np.unique(sorted(discontinuous_indices))
+            ), "Indices must me monotonic."
+            interval_indices = [
+                [discontinuous_indices[i - 1], discontinuous_indices[i]]
+                for i in range(len(discontinuous_indices))
+            ]
+
         self._knots = knots
-        self.method = method
+        self._method = method
+        self.intervals = interval_indices
```

Comment:
Can you explain to me why you needed to add the new attributes `discontinuous_indices` and `intervals` and how they are actually used? I don't see where they get referenced in these changes.

Reply:
The user inputs indices at which they want the curve to be discontinuous (`discontinuous_indices`). These indices are taken and are used to create the separate intervals of the curve (`self.intervals`). e.g. if `discontinuous_indices = [0, 7]`, then `self.intervals = [[0, 7], [7, 0]]`. The intervals are then iterated through in the compute functions  -- via transforms -- such that each interval is interpolated separately.

Is that what you're looking for?